### DISCUSSION TITLE
How to pass parse_float option to DRF json decoder

### DISCUSSION THREAD

**Discussion by 0xNima** [encode/django-rest-framework#8488] [Question & Answer] [answered]
Hi. Is there any convenient way to pass the `parse_float` option to the JSON decoder?
When I send a JSON body like:
```
data = "{\"amount\": 1.000000000000002345}"
```
`request.data` in view class give me:

`{'amount': 1.0000000000000024}`

 but if decode it as `json.loads(data, parse_float=Decimal)` it will be fine.
 
 I'm looking for a straightforward solution rather than defining a custom JSON parser. Like pass it from view as a class variable and use it in the parser

**Comment by yyyyyyyan** [comment, marked as answer]
Hi! The DRF builtin `JSONParser.parse()` always calls `json.load()` with two arguments - the data to be loaded and `parse_constant`. You can either create a custom Parser or send the decimal number as a string you can properly convert to a Decimal with a serializer.

**Comment by hudy0** [comment]
https://www.django-rest-framework.org/api-guide/parsers/
